### PR TITLE
[Merged by Bors] - More api changes

### PR DESCRIPTION
**Pull request by acrrd**
Depends on #741 
* Document ingestion now returns 201 instead of 204; error code 422 has been replaced with 409.
* Request bodies have been marked as required.
* A response always returns a body, even when empty.
* 404 is returned only if the request does not enter the handler. If it does, then a 400 will be returned instead.
* Now every error is a `GenericError`
* Specific error are now only one possible error that the endpoint can return, a generic error is always possible. In this way we can extend the kind of error we can return form an endpoint.

### CODE REVIEW

File: web-api/openapi/user_management.yaml [hunk 1]
```diff
@@ -138,11 +145,13 @@ components:
...
-        - $ref: './schemas/error.yml#/BaseError'
-        - type: object
-          required: [kind]
-          properties:
-            kind:
-              type: string
-              enum: [InvalidUserId, InvalidDocumentId]
+      oneOf:
+        - $ref: './schemas/error.yml#/GenericError'
+        - allOf:
```
Comment (rustonaut) [inline web-api/openapi/user_management.yaml:150]: Depending on the code generation we might need to switch the order of the `oneOf` around to go from most-precise to least-precise.

**Review by rustonaut — APPROVED** [review]
lgtm, through we might need one follow up depending on code gen (it might also be right just the way it is now, or it might not matter at all)

File: web-api/openapi/user_management.yaml (hunk 1 above)
Comment (acrrd) [reply web-api/openapi/user_management.yaml:150]: @frankpepermans

Comment (acrrd) [reply web-api/openapi/user_management.yaml]: I removed this commit because it breaks the generation in all targets :(

**Comment by acrrd** [comment]
I reverted the change on the errors because it broke code generation.

bors merge

**Pull request closed without merging**